Update manual for Smart Proxy puppetca and the new providers

dgoetz opened the pull request:
Update of the documentation was requested by @tbrisker in https://github.com/theforeman/smart-proxy/pull/586. @timogoebel can you please review?

Should the changes also be backported to older manuals? If yes, 1.19 included the change to make puppetca plug-able and 1.20 added the token based mechanism, correct?

dgoetz committed:
Update manual for Smart Proxy puppetca and the new providers

tbrisker commented:
Thanks for the PR @dgoetz ! Would be good to update the manuals for 1.19 and 1.20 as well since they are still supported.

dgoetz commented:
Ok, I wait for a review on the content and then backport to the older ones. So please do not merge before.

timogoebel commented on _includes/manuals/nightly/4.3.7_smartproxy_puppetca.md:
```diff
@@ -1,23 +1,40 @@
 :autosignfile: /etc/puppet/autosign.conf
 </pre>
 
+The location of the file you can determine with `puppet config print autosign`.
```
... `can be determined with`

timogoebel reviewed (commented):
Just one nit. Otherwise: Looks good.

ekohl commented:
This looks like a very good start, but it's important to also note you need to install the script to do verification. https://github.com/theforeman/smart-proxy/blob/develop/extra/puppet_sign.rb should be deployed somewhere and puppet's autosign setting should point to it. By default it's installed to `/usr/share/foreman-proxy/extra/puppet_sign.rb`. Not sure if permissions need to be fixed as well.

dgoetz commented:
Thanks for the reviews so far.
@timogoebel: I fixed the wording.
@ekohl: I added an additional section for the Puppet part of the token based mechanism.
So if you both can please re-review.

timogoebel commented:
@leewaa: Do you mind reviewing as well (especially the English 🇦🇺)?

leewaa commented:
Sure

leewaa commented:
LGTM

ekohl reviewed (approved):
Looks correct to me

tbrisker commented:
@timogoebel @leewaa  @ekohl  - is this all valid for both 1.19 and 1.20? or do we need some changes for one or both?

dgoetz commented:
The token-based provider is only in 1.20, so I would backport the complete text to 1.20 and only the first half to 1.19.

dgoetz commented:
Backports are done and tests run successfully so feel free to merge from my side.

tbrisker reviewed (approved):
Thank you very much @dgoetz and all reviewers!

ekohl commented:
Now that I dive a bit deeper into this: We install the puppet_sign.rb script twice. The one in `/usr/libexec/foreman-proxy/puppet_sign.rb` is executable while the one in `/usr/share` isn't.

dgoetz commented:
@ekohl: Should I create a follow-up pull-request or do you want to do so?

ekohl commented:
@dgoetz be my guest. Then I'll continue preparing the 1.21 release :)
